### PR TITLE
lint: Update golangci-lint and fix gocyclo issues reported by golangci- lint

### PR DESCRIPTION
**Pull request by tariq1890**
Fix golangci-lint  issues in the repo. Let's enforce some code quality :) !

```release-note
Update Golangci-lint version and fix linting issues reported by golangci-lint.
```

This PR includes the following changes

i) Add bazel CI script to run the linter.
ii) Add `gocyclo` settings to the GolangCI yaml.
iii) Fix the `gocyclo` failure in `virtualmachines.go`.
`Line 76: warning: cyclomatic complexity 18 of function (*Service).createVM() is high (> 15) (gocyclo)`

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -21,4 +24,8 @@ linters:
   fast: true
 issue:
   max-same-issues: 0
-max-per-linter: 0
\ No newline at end of file
+max-per-linter: 0
+
+run:
+  skip-dirs:
+  - build
```
Comment (soggiest) [inline .golangci.yml]: Need a newline here

File: pkg/cloud/azure/services/compute/virtualmachines.go [hunk 2]
```diff
@@ -78,31 +78,91 @@ func (s *Service) createVM(machine *actuators.MachineScope, bootstrapToken, kube
...
+
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to lookup machine %q", machine.Name())
+	}
+	return instance != nil, nil
+}
+
+func (s *Service) getVMStartupScript(machine *actuators.MachineScope, bootstrapToken string) (string, error) {
+	var startupScript string
+
```
Comment (soggiest) [inline pkg/cloud/azure/services/compute/virtualmachines.go]: This looks like it belongs in a different PR than one for updating gloangci-lint

File: scripts/ci-bazel-lint.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+cd $REPO_ROOT
+golangci-lint run
```
Comment (soggiest) [inline scripts/ci-bazel-lint.sh]: Newline here too

File: .golangci.yml (hunk 1 above)
Comment (soggiest) [inline .golangci.yml]: Newline here

File: pkg/cloud/azure/services/compute/virtualmachines.go [hunk 4]
```diff
@@ -143,83 +203,28 @@ func (s *Service) createVM(machine *actuators.MachineScope, bootstrapToken, kube
...
-
-	if machine.MachineStatus.VMID != nil {
-		instance, err = s.VMIfExists(machine.Name())
-	}
-
-	if err != nil {
-		return false, errors.Wrapf(err, "failed to lookup machine %q", machine.Name())
-	}
-	return instance != nil, nil
+	return startupScript, nil
```
Comment (soggiest) [inline pkg/cloud/azure/services/compute/virtualmachines.go]: This needs to be moved into its own PR, this doesnt look like it has to do with golangci-lint

File: scripts/ci-bazel-lint.sh (hunk 3 above)
Comment (soggiest) [inline scripts/ci-bazel-lint.sh]: Newline needed here too

**Comment by justaugustus** [comment]
/hold
(For my review as well)

File: pkg/cloud/azure/services/compute/virtualmachines.go (hunk 4 above)
Comment (tariq1890) [reply pkg/cloud/azure/services/compute/virtualmachines.go]: This is needed because this is a piece of code the violates the `gocyclo` linter. I am using this PR so that the project becomes ready for a CI lint job. So I'd argue that this change is still in scope of what this PR intends to do.

Comment (soggiest) [reply pkg/cloud/azure/services/compute/virtualmachines.go]: I see, that wasn't obvious to me. Thank you for the clarification
